src/transform/ir_proto_to_datalog/arcs_manifest_tree/type_test.cc

```diff
@@ -0,0 +1,231 @@
+#include "src/transform/ir_proto_to_datalog/arcs_manifest_tree/type.h"
+
+#include <algorithm>
+#include <memory>
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/strings/str_split.h"
+#include "src/ir/access_path_selectors.h"
+#include "src/ir/access_path_selectors_set.h"
+#include "src/common/logging/logging.h"
+#include "src/common/testing/gtest.h"
+#include "src/transform/ir_proto_to_datalog/arcs_manifest_tree/entity_type.h"
+#include "src/transform/ir_proto_to_datalog/arcs_manifest_tree/primitive_type.h"
+#include "src/transform/ir_proto_to_datalog/arcs_manifest_tree/schema.h"
+
+namespace raksha::transform::arcs_manifest_tree {
+
+// Helper function for making an unnamed schema from a field map.
+static Schema MakeAnonymousSchema(
+    absl::flat_hash_map<std::string, std::unique_ptr<Type>> field_map) {
+  return Schema(absl::optional<std::string>(), std::move(field_map));
+}
+
+// Helper function for making an entity type with an unnamed schema from a
+// field map.
+static EntityType MakeEntityTypeWithAnonymousSchema(
+    absl::flat_hash_map<std::string, std::unique_ptr<Type>> field_map) {
+  return EntityType(MakeAnonymousSchema(std::move(field_map)));
+}
+
+static std::unique_ptr<EntityType> MakeUniquePtrEntityTypeWithAnonymousSchema(
+    absl::flat_hash_map<std::string, std::unique_ptr<Type>> field_map) {
+  return std::make_unique<EntityType>(
+      MakeAnonymousSchema(std::move(field_map)));
+}
+
+// Given a vector of strings representing access paths to leaves, generate a
+// type having all of those access paths and no others.
+// The resulting type will not have information about the name of the type;
+// this is not presented by the selector access path and thus cannot be
+// derived from it.
+template<typename Iter>
```

Comment:
It seems like a very complicated logic to use for tests. It also makes the test harder to read. Let us leave this for now. However, I think we need a better mechanism to write these tests. I can think of a few options:
 - Use the `.arcs` format to write schemas, parse them as protos, and use in tests. (This requires the code to deserialize the `TypeProto` to `Type`).
 - Use something like json to express the nesting of fields and use that to generate the desired type. 
 - Create something like `class FieldMap(std::map<std::string, std::unique_ptr<FieldMap>)` with appropriate factory methods to make it easy to create types. 

The first option might be reasonable, but the only issue is that you will need to create `.arcs` files instead of using inline strings.  Can you file a bug?

Reply:
We could also use inline `.textproto` strings. I've done some of that in the follow-up PR #87. Perhaps these should just be moved to the test infrastructure I've put up there and we can get rid of these complex helper methods.

Reply:
Issue opened as #88.